Mathlib/Data/Rel.lean

```diff
@@ -262,8 +355,42 @@ namespace Function
 def graph (f : α → β) : Rel α β := fun x y => f x = y
 #align function.graph Function.graph
 
+theorem graph_id : graph id = @Eq α := by simp[graph]
+
+theorem graph_comp {f : β → γ} {g : α → β} : graph (f ∘ g) = Rel.comp (graph g) (graph f) := by
+  ext x y
+  simp[graph, Rel.comp]
+
 end Function
 
+theorem Equiv.graph_inv (f : α ≃ β) : f.invFun.graph = Rel.inv (f.toFun.graph) := by
+  ext x y
+  simp[Function.graph, Rel.inv, Equiv.apply_eq_iff_eq_symm_apply (f := f.symm), flip]
+  exact eq_comm
+
+theorem Relation.is_graph_iff (r : Rel α β): (∃! f, Function.graph f = r) ↔ ∀ x, ∃! y, r x y := by
+  unfold Function.graph
+  constructor
+  · rintro ⟨f, hf⟩ x
+    use f x
+    rw[←hf.left]
+    simp only [forall_eq', and_self]
```

Comment:
A neat trick is: if you have `hf : f x = r` in your hypotheses, `rcases hf with rfl` will replace `r` with `f x` everywhere. So in `rcases` syntax we get:
```suggestion
  · rintro ⟨f, rfl, _⟩ x
    use f x
    simp only [forall_eq', and_self]
```